src/core/ext/filters/client_channel/lb_policy/grpclb/load_balancer_api.c

```diff
@@ -258,13 +258,10 @@ int grpc_grpclb_duration_compare(const grpc_grpclb_duration *lhs,
   return 0;
 }
 
-gpr_timespec grpc_grpclb_duration_to_timespec(
-    grpc_grpclb_duration *duration_pb) {
-  gpr_timespec duration;
-  duration.tv_sec = duration_pb->has_seconds ? duration_pb->seconds : 0;
-  duration.tv_nsec = duration_pb->has_nanos ? duration_pb->nanos : 0;
-  duration.clock_type = GPR_TIMESPAN;
-  return duration;
+grpc_millis grpc_grpclb_duration_to_millis(grpc_grpclb_duration *duration_pb) {
+  return (duration_pb->has_seconds ? duration_pb->seconds : 0) *
+             GPR_MS_PER_SEC +
+         (duration_pb->has_nanos ? duration_pb->nanos : 0) / GPR_NS_PER_MS;
```

Comment:
This is a nit, but let's round down: `duration` represents how long a serverlist is valid. Rounding down is slightly more conservative.